src/api/mediators.js

```diff
@@ -0,0 +1,456 @@
+		return;
+	}
+
+	try {
+		const m = yield Mediator.find().exec();
+		maskPasswords(m.configDefs, m.config);
+		return this.body = m;
+	} catch (err) {
+		return utils.logAndSetResponse(this, 500, `Could not fetch mediators via the API: ${err}`, "error");
+	}
+}
+
+
+export function* getMediator(mediatorURN) {
+	// Must be admin
+	if (!authorisation.inGroup("admin", this.authenticated)) {
+		utils.logAndSetResponse(this, 403, `User ${this.authenticated.email} is not an admin, API access to getMediator denied.`, "info");
+		return;
+	}
+
+	const urn = unescape(mediatorURN);
+
+	try {
+		const result = yield Mediator.findOne({ urn }).exec();
+		if (result === null) {
+			return this.status = 404;
+		} else {
+			maskPasswords(result.configDefs, result.config);
+			return this.body = result;
+		}
+	} catch (err) {
+		return utils.logAndSetResponse(this, 500, `Could not fetch mediator using UUID ${urn} via the API: ${err}`, "error");
+	}
+}
+
+function constructError(message, name) {
+	const err = new Error(message);
+	err.name = name;
+	return err;
+}
+
+
+function validateConfigDef(def) {
+	if (def.type === "struct" && !def.template) {
+		throw constructError(`Must specify a template for struct param '${def.param}'`, "ValidationError");
+	} else if (def.type === "struct") {
+		for (const templateItem of Array.from(def.template)) {
+			if (!templateItem.param) {
+				throw constructError(`Must specify field 'param' in template definition for param '${def.param}'`, "ValidationError");
+			}
+
+			if (!templateItem.type) {
+				throw constructError(`Must specify field 'type' in template definition for param '${def.param}'`, "ValidationError");
+			}
+
+			if (templateItem.type === "struct") {
+				throw constructError(`May not recursively specify 'struct' in template definitions (param '${def.param}')`, "ValidationError");
+			}
+		}
+	} else if (def.type === "option") {
+		if (!utils.typeIsArray(def.values)) {
+			throw constructError(`Expected field 'values' to be an array (option param '${def.param}')`, "ValidationError");
+		}
+		if ((def.values == null) || (def.values.length === 0)) {
+			throw constructError(`Must specify a values array for option param '${def.param}'`, "ValidationError");
+		}
+	}
+}
+
+// validations additional to the mongoose schema validation
+const validateConfigDefs = configDefs => Array.from(configDefs).map((def) => validateConfigDef(def));
+
+
+export function* addMediator() {
+	// Must be admin
+	if (!authorisation.inGroup("admin", this.authenticated)) {
+		utils.logAndSetResponse(this, 403, `User ${this.authenticated.email} is not an admin, API access to addMediator denied.`, "info");
+		return;
+	}
+
+	try {
+		let mediatorHost;
+		const mediator = this.request.body;
+
+		if (__guard__(__guard__(mediator != null ? mediator.endpoints : undefined, x1 => x1[0]), x => x.host) != null) {
```

Comment:
Removed all the guards for plain js. It's not as concise as it could be but it's easy to understand.